scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/package.scala

```diff
@@ -0,0 +1,118 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.parquet
+
+import java.lang.{Boolean => JBoolean}
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.testing.AvroIO
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import org.apache.avro.Schema
+import org.apache.avro.specific.SpecificRecordBase
+import org.apache.beam.sdk.io.hadoop.inputformat.HadoopInputFormatIO
+import org.apache.beam.sdk.transforms.SimpleFunction
+import org.apache.hadoop.mapreduce.Job
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
+import org.apache.parquet.avro.AvroParquetInputFormat
+import org.apache.parquet.filter2.predicate.FilterPredicate
+import org.apache.parquet.hadoop.ParquetInputFormat
+
+import scala.reflect.ClassTag
+
+/**
+ * Main package for Parquet Avro APIs. Import all.
+ *
+ * {{{
+ * import com.spotify.scio.parquet.avro._
+ * }}}
+ */
+package object avro {
+
+  /** Alias for [[me.lyh.parquet.avro.Projection]]. */
+  val Projection = me.lyh.parquet.avro.Projection
+
+  /** Alias for [[me.lyh.parquet.avro.Predicate]]. */
+  val Predicate = me.lyh.parquet.avro.Predicate
+
+  /** Enhanced version of [[ScioContext]] with Parquet Avro methods. */
+  implicit class ParquetAvroScioContext(val self: ScioContext) extends AnyVal {
+
+    /**
+     * Get an SCollection for a Parquet file as Avro records.
+     *
+     * Note that due to limitations of the underlying [[HadoopInputFormatIO]],
+     * Avro [[org.apache.avro.generic.GenericRecord GenericRecord]] and dynamic work rebalancing
+     * are not supported. Without the latter, pipelines may not autoscale up or down during the
+     * initial read and subsequent fused transforms.
+     *
+     * @group input
+     */
+    def parquetAvroFile[T <: SpecificRecordBase : ClassTag](path: String,
+                                                            projection: Schema = null,
+                                                            predicate: FilterPredicate = null)
+    : SCollection[T] = self.requireNotClosed {
+      if (self.isTest) {
+        self.getTestInput(AvroIO[T](path))
+      } else {
+        val cls = ScioUtil.classOf[T]
+        val readSchema = cls.getMethod("getClassSchema").invoke(null).asInstanceOf[Schema]
+
+        val job = Job.getInstance()
+
+        // Client id/secret of Google-managed project associated with the Cloud SDK
+        // These are needed since `FileInputFormat.setInputPaths` validates paths locally and
+        // requires the user's GCP credentials.
+        job.getConfiguration.setBoolean("fs.gs.auth.service.account.enable", false)
```

Comment:
IRL. we could discover `GOOGLE_APPLICATION_CREDENTIALS` and configure it via https://github.com/GoogleCloudPlatform/bigdata-interop/blob/5f1138979919c9cc38f5dfcc4d7e73ffa7b5e176/util/src/main/java/com/google/cloud/hadoop/util/EntriesCredentialConfiguration.java#L60-L76